XmlFormatter/src/Windows/MainForm.cs

```diff
@@ -191,104 +190,155 @@ private void MainForm_DragDrop(object sender, DragEventArgs e)
         {
-            stringVersion += ".0";
-            Regex regex = new Regex(@"[0-9]{1,}.[0-9]{1,}.[0-9]{1,}");
-            if (!regex.IsMatch(stringVersion))
-            {
-                stringVersion = "0.0.0.0";
-            }
-            Version version = new Version(stringVersion);
-
-            return version;
+            MessageBox.Show(e.Message, e.Title, MessageBoxButtons.OK, MessageBoxIcon.Error);
         }
 
         /// <summary>
-        /// This method will return you the string of a given version
+        /// Click on the report bug button
         /// </summary>
-        /// <param name="version">The version to get the string for</param>
-        /// <returns>A string representatiion of the given version</returns>
-        private string GetStringVersion(Version version)
+        /// <param name="sender">The sender of the event</param>
+        /// <param name="e">The arguments from the sender</param>
+        private void MI_ReportIssue_Click(object sender, EventArgs e)
         {
-            return version.Major + "." + version.Minor + "." + version.Build;
+            Process.Start("https://github.com/XanatosX/XmlFormatter/issues");
         }
 
-
         /// <summary>
-        /// This method will check if there is newer version for download
+        /// The close event of the form
         /// </summary>
-        /// <param name="sender">The control triggering the event</param>
-        /// <param name="e">The event data</param>
-        private async void MI_CheckForUpdate_Click(object sender, EventArgs e)
+        /// <param name="sender">The sender of the event</param>
+        /// <param name="e">The arguments from the sender</param>
+        private void MainForm_FormClosing(object sender, FormClosingEventArgs e)
         {
-            GitHubClient client = new GitHubClient(new ProductHeaderValue("XanatosX"));
-            var releases = await client.Repository.Release.GetAll("XanatosX", "XmlFormatter");
-            if (releases.Count == 0)
+            if (Properties.Settings.Default.AskBeforeClosing)
             {
-                MessageBox.Show("Could't not find a version onn GitHub", "No version found");
-            }
-            Release latestRelease = releases[0];
-            string latest = latestRelease.Name;
-            Regex regex = new Regex(@"([0-9]{1,}.[0-9]{1,}.[0-9]{1,})");
-            Version gitHubVersion = null;
-            foreach (Match match in regex.Matches(latest))
-            {
-                gitHubVersion = ConvertInnerFormatToProperVersion(match.Value);
+                DialogResult result = MessageBox.Show(
+                    "Do you really want to close this application?",
+                    "Close the application",
+                    MessageBoxButtons.YesNo,
+                    MessageBoxIcon.Question
+                );
+
+                if (result != DialogResult.Yes)
+                {
+                    e.Cancel = true;
+                    return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this redundant jump.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=4995221)